src/rust/lib/graph-editor/src/component/visualization/data.rs

```diff
@@ -0,0 +1,107 @@
+//! This module defines the `Data` struct and related functionality.
+
+use crate::prelude::*;
+
+use crate::component::visualization::EnsoType;
+
+use serde::Deserialize;
+
+
+
+// ======================================
+// === Wrapper for Visualisation Data ===
+// =======================================
+
+/// Type indicator
+pub type DataType = EnsoType;
+
+/// Wrapper for data that can be consumed by a visualisation.
+/// TODO[mm] consider static versus dynamic typing for visualizations and data!
+#[derive(Clone,CloneRef,Debug)]
+#[allow(missing_docs)]
+pub enum Data {
+    JSON   { content : Rc<serde_json::Value> },
+    Binary { content : Rc<dyn Any>           },
```

Comment:
Its ok to leave it this way, but pls add todo comment - we should choose a concrete type here which will mean that it is a binary stream.